### PR TITLE
Code Quality Updates - Part 2

### PR DESCRIPTION
**Pull request by jbouder**
## Reference Issues or PRs

N/A

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

## Any other comments?
- Misc styling fixes
- Fix unit test console errors
- Add unit tests
- Remove jest dev environment config

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "lint": "eslint src --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
     "preview": "vite preview",
     "format": "npx prettier src --write",
-    "test": "NODE_ENV=development jest",
```
Comment (jbouder) [inline ui/package.json:15]: This was something we unnecessarily added when implementing the dummy axios data, to support testing in vercel. It actually is not needed, and is resulting in a lot of unnecessary errors to be logged when running unit tests.

File: ui/src/components/app-card/app-card.test.tsx [hunk 2]
```diff
@@ -30,7 +30,6 @@ describe('AppCard', () => {
             username="Developer"
             framework="Some Framework"
             url="/some-url"
-            ready={true}
```
Comment (jbouder) [inline ui/src/components/app-card/app-card.test.tsx:33]: This is no longer used in the card, removing

File: ui/src/pages/server-types/server-types.tsx [hunk 3]
```diff
@@ -210,9 +221,7 @@ export const ServerTypes = (): React.ReactElement => {
           .
         </p>
       </div>
-      {error ? (
-        <div>An error occurred: {(error as Error).message}</div>
-      ) : isLoading ? (
```
Comment (jbouder) [inline ui/src/pages/server-types/server-types.tsx:215]: Removed and replaced with current pattern or displaying overall errors at app level

**Review by kildre — APPROVED** [review]

**Review by aktech — APPROVED** [review]
Approved on the basis on @kildre's approval

**Pull request merged**